manifests/init.pp

```diff
@@ -99,25 +99,41 @@
   }
 
   if $rh_user {
-    $_user = " --username='${rh_user}'"
+    if $rh_user.is_a(Deferred) {
+      $_user = Deferred('inline_epp', [' --username="<%= $rh_user %>"', { 'rh_user' => $rh_user }])
+    } else {
+      $_user = " --username='${rh_user}'"
+    }
```

Comment:
We can eliminate the repetition of the variable assignment with these if/else statements:

```suggestion
    $_user = if $rh_user.is_a(Deferred) {
      Deferred('inline_epp', [' --username="<%= $rh_user %>"', { 'rh_user' => $rh_user }])
    } else {
      " --username='${rh_user}'"
    }
```